services/web/server/src/simcore_service_webserver/login/utils.py

```diff
@@ -127,6 +127,8 @@ async def send_mail(app: web.Application, msg: MIMEText):
         use_tls=cfg.SMTP_TLS_ENABLED,
     )
     log.debug("Sending email with smtp configuration: %s", pformat(smtp_args))
+    assert cfg.SMTP_PASSWORD  # nosec
+    assert cfg.SMTP_USERNAME  # nosec
```

Comment:
@pcrespov @sanderegg user and password are optional in the configuration, but always required here. Any reason for this? can we just make them required?

Reply:
This assert is wrong. Look in line 141.  That is why they are optional

I suggest to add a validation check in the ``SMTPSettings`` settings to ensure that sure that ``SMTP_TLS_ENABLED``, ``SMTP_USERNAME`` and ``SMTP_PASSWORD`` are consistent. e.g. if tls is enabled, then user and password is required. If user is passed, perhaps password should be required etc

Reply:
Good one. Will remove the asserts and add validators.